resharper/resharper-unity/src/CSharp/Feature/Services/LiveTemplates/Scope/UnityProjectVersionScopeProvider.cs

```diff
@@ -20,28 +19,22 @@ public IEnumerable<ITemplateScopePoint> ProvideScopePoints(TemplateAcceptanceCon
                 yield break;
 
             var project = context.GetProject();
-            if (project != null && !project.IsUnityProject())
-                yield break;
+            var version = project != null
+                ? context.Solution.GetComponent<UnityVersion>().GetActualVersion(project)
+                : context.Solution.GetComponent<UnityVersion>().ActualVersionForSolution.Value;
```

Comment:
I would rather check ActualVersionForSolution first.

            var version = project != null
                ? context.Solution.GetComponent<UnityVersion>().ActualVersionForSolution.Value;
                : context.Solution.GetComponent<UnityVersion>().GetActualVersion(project)

Reply:
I think it's fine for now. We're using `GetActualVersion(IProject)` in lots of places, including `UnityApi` for completion, usage suppression and so on. I'm up for reviewing `UnityVersion` usage in 211 though - it might be worth removing the `IProject` API completely.